tests/integration/src/test/java/org/mule/test/config/ExceptionHelperTestCase.java

```diff
@@ -80,9 +85,70 @@ public void testGetNonMuleExceptionCause()
         assertThat(ExceptionHelper.getNonMuleException(new IllegalArgumentException()),IsInstanceOf.instanceOf(IllegalArgumentException.class));
     }
 
+    @Test
+    public void testFilteredStack1()
+    {
+        int calls = 5;
+        try
+        {
+            recursive(calls, new Closure()
+            {
+                @Override
+                public void execute(Object input)
+                {
+                    CollectionUtils.forAllDo(Collections.singleton(null), new Closure()
+                    {
+                        @Override
+                        public void execute(Object input)
+                        {
+                            throw new RuntimeException(new DefaultMuleException(MessageFactory.createStaticMessage("foo")));
+                        }
+                    });
+                }
+            });
+        }
+        catch (Exception e)
+        {
+            assertThat(ExceptionHelper.getExceptionStack(e), containsString("foo (org.mule.api.DefaultMuleException)"));
+            assertThat(ExceptionHelper.getExceptionStack(e), containsString("  org.mule.test.config.ExceptionHelperTestCase$1$1:"));
+            assertThat(ExceptionHelper.getExceptionStack(e), containsString("  (1 more...)")); // CollectionUtils.forAllDo
+            assertThat(ExceptionHelper.getExceptionStack(e), containsString("  org.mule.test.config.ExceptionHelperTestCase$1:"));
+            assertThat(ExceptionHelper.getExceptionStack(e), containsString("  org.mule.test.config.ExceptionHelperTestCase:"));
+            assertThat(ExceptionHelper.getExceptionStack(e), containsString("  (" + (calls + 13) + " more...)")); // recursive
```

Comment:
Can you create a matcher that receives a the expected strings and compares that against the exception's stack trace?